darwin/future/meta/objects/item.py

```diff
@@ -53,6 +54,18 @@ def delete(self) -> None:
         filters = {"item_ids": [str(self.id)]}
         delete_list_of_items(self.client, team_slug, dataset_id, filters)
 
+    def restore(self) -> None:
+        team_slug, dataset_id = (
+            self.meta_params["team_slug"],
+            self.meta_params["dataset_id"]
+            if "dataset_id" in self.meta_params
+            else self.meta_params["dataset_ids"],
+        )
+        assert isinstance(team_slug, str)
+        dataset_id = cast(Union[int, List[int]], dataset_id)
```

Comment:
I'm a fan of your use of `cast` in these tickets John, assuming we're confident of the typing in `meta_params`